src/ui/public/index_patterns/validate/validate_index_pattern.js

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { INDEX_PATTERN_ILLEGAL_CHARACTERS_VISIBLE } from '../constants';
+
+export function validateIndexPattern(indexPattern, fieldName = 'index pattern') {
```

Comment:
After talking things over with @bmcconaghy we agreed that the consumers may want control over what types of messages they show the user, depending on context. So I'm going to refactor this a bit to only encapsulate the validation logic and leave it to the consumer to decide what to do based on the validation outcome.